internal/server/iifo.go

```diff
@@ -46,8 +48,13 @@ func (s *SOMASServer) getPredictions() shared.DisasterPredictionInfoDict {
 	nonDeadClients := getNonDeadClientIDs(s.gameState.ClientInfos)
 	for _, id := range nonDeadClients {
 		c := s.clientMap[id]
-		islandPredictionsDict[id] = c.MakeDisasterPrediction()
+		disasterPrediction := c.MakeDisasterPrediction()
+		if math.IsNaN(disasterPrediction.PredictionMade.Confidence) || disasterPrediction.PredictionMade.TimeLeft == -9223372036854775808 || math.IsNaN(disasterPrediction.PredictionMade.Magnitude) || math.IsNaN(disasterPrediction.PredictionMade.CoordinateY) || math.IsNaN(disasterPrediction.PredictionMade.CoordinateX) {
```

Comment:
IMO we should not _produce_ `NaN`s/or min value. this should be handled at creation time, but this gets values from client side so here is fine.

Reply:
see https://stackoverflow.com/questions/6878590/the-maximum-value-for-an-int-type-in-go for solution

Reply:
Should I make <math.MinINT32 or something then?

Reply:
I feel like `TimeLeft` should be a `uint`? then we don't really need to do anything about it.

Reply:
Made it a uint but it could still be overflowed... 
````
2021/01/07 17:29:51 [Team6]: Final Prediction: [{NaN NaN NaN 9223372036854775808 NaN}]
2021/01/07 17:29:51 [Team1]: Final Prediction: [{1.2 1.2 1.2 1 99}]
2021/01/07 17:29:51 [Team2]: Final Prediction: [{1.2 1.2 1.2 1 99}]
2021/01/07 17:29:51 [Team3]: Final Prediction: [{NaN NaN NaN 9223372036854775808 NaN}]
2021/01/07 17:29:51 [Team4]: Final Prediction: [{1.2 1.2 1.2 1 99}]
2021/01/07 17:29:51 [Team5]: Final Prediction: [{1.2 1.2 1.2 1 99}]
````

Reply:
The final prediction is not used anywhere by team3 and team6 yet and is never distributed so its not breaking the codebase AFAIK